workitem/workitem_repository.go

```diff
@@ -690,48 +693,175 @@ func (r *GormWorkItemRepository) Fetch(ctx context.Context, spaceID uuid.UUID, c
+	db := r.db.Model(&iteration.Iteration{}).Where("space_id = ?", spaceID)
+	if db.Error != nil {
+		return nil, errors.NewInternalError(db.Error.Error())
+	}
+	var allIterations []uuid.UUID
+	db.Pluck("id", &allIterations)
+	iterationTable := iteration.Iteration{}
+	iterationTableName := iterationTable.TableName()
+	iterationWithWICount := fmt.Sprintf(`
+	SELECT count(*) AS Total,
+		count(CASE fields->>'system.state'
+					WHEN 'closed' THEN '1'
+					ELSE NULL
+				END) AS Closed,
+		fields->>'system.iteration' AS iterationID
+	FROM %s wi
+	WHERE fields->>'system.iteration' IN
+		(SELECT id::text
+		FROM %s
+		WHERE space_id='%s')
+	AND wi.deleted_at IS NULL
+	GROUP BY fields->>'system.iteration'`,
+		workitemTableName,
+		iterationTableName,
+		spaceID.String())
+	db = r.db.Raw(iterationWithWICount)
 	var res []WICountsPerIteration
-	db := r.db.Table("work_items").Select(`iterations.id as IterationId, count(*) as Total,
-				count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed`).Joins(`left join iterations
-				on fields@> concat('{"system.iteration": "', iterations.id, '"}')::jsonb`).Where(`iterations.space_id = ?
-				and work_items.deleted_at IS NULL`, spaceID).Group(`IterationId`).Scan(&res)
+	db.Scan(&res)
 	if db.Error != nil {
+		log.Error(ctx, map[string]interface{}{
+			"space_id": spaceID.String(),
+			"err":      db.Error,
+		}, "unable to count WI for every iteration in a space")
 		return nil, errors.NewInternalError(db.Error.Error())
 	}
+	wiMap := map[string]WICountsPerIteration{}
+	for _, r := range res {
+		wiMap[r.IterationID] = WICountsPerIteration{
+			IterationID: r.IterationID,
+			Total:       r.Total,
+			Closed:      r.Closed,
+		}
+	}
+	// put 0 count for iterations which are not in wiMap
+	// ToDo: Update count query to include non matching rows with 0 values
+	// Following operation can be skipped once above is done
+	for _, i := range allIterations {
+		if _, exists := wiMap[i.String()]; exists == false {
+			wiMap[i.String()] = WICountsPerIteration{
+				IterationID: i.String(),
+				Total:       0,
+				Closed:      0,
+			}
+		}
+	}
+
+	type IterationHavingChildrenID struct {
+		Children    string `gorm:"column:children"`
+		IterationID string `gorm:"column:iterationid"`
+	}
+	var itrChildren []IterationHavingChildrenID
+	queryIterationWithChildren := fmt.Sprintf(`
+	WITH PathResolver AS
+	(SELECT CASE
+				WHEN path = '' THEN replace(id::text, '-', '_')::ltree
+				ELSE concat(path::text, '.', REPLACE(id::text, '-', '_'))::ltree
+			END AS pathself,
+			id
+	FROM %s)
+	SELECT array_agg(iterations.id)::text AS children,
+		PathResolver.id::text AS iterationid
+	FROM %s,
+		PathResolver
+	WHERE path <@ PathResolver.pathself
+	AND space_id = '%s'
+	GROUP BY (PathResolver.pathself,
+		PathResolver.id)`,
+		iterationTableName,
+		iterationTableName,
+		spaceID.String())
+	db = r.db.Raw(queryIterationWithChildren)
```

Comment:
Same here, please use `?` instead of `%s` and don't use `fmt.Sprintf` for formatting SQL strings.